pkg/internal/discover/watcher_kube.go

```diff
@@ -73,7 +73,7 @@ func (wk *watcherKubeEnricher) ID() string { return "unique-watcher-kube-enriche
 // handling in the enrich main loop
 func (wk *watcherKubeEnricher) On(event *informer.Event) {
 	// ignoring updates on non-pod resources
-	if event.Resource.Pod == nil {
+	if event == nil || event.Resource == nil || event.Resource.Pod == nil {
```

Comment:
When you rebase, you will have a conflict, as this has been already fixed in main: https://github.com/grafana/beyla/blob/main/pkg/internal/discover/watcher_kube.go#L75